### PR TITLE
CMake Build System

### LINKED ISSUE #55: Consider: Remove Batch File and Makefile and Switch Entirely to CMake
I'm opening this issue to discuss removing the included Makefile (for Linux) and batch file (for Windows) and switching entirely to a CMake build system instead.  This is inspired by several related issues and pull requests.  For example:

* https://github.com/KhronosGroup/OpenCL-ICD-Loader/pull/22
* https://github.com/KhronosGroup/OpenCL-ICD-Loader/pull/26
* https://github.com/KhronosGroup/OpenCL-ICD-Loader/pull/43

Since the Makefile and batch file use CMake anyhow, can we simply remove them and update the README to describe how to use CMake directly instead?

### PR DESCRIPTION
**Pull request by bashbaug**
This is a WIP pull request for discussion.  It is one way to resolve #55.  It removes the build batch files and Makefiles and instead describes how to build using CMake in the README.

### CODE REVIEW

**Comment by bashbaug** [comment]
I believe this change is now ready to go.

**Comment by bashbaug** [comment]
@kepatil @alycm  could one of you please review and merge these changes?

Other than removing the Makefiles there are no functional changes, but it'd be worth reviewing the README changes to be sure they are accurate.  Thanks!

File: README.md [hunk 1]
```diff
@@ -1,49 +1,114 @@
...
+To use system OpenCL Headers, please specify the OpenCL Header location using the CMake variable `OPENCL_ICD_LOADER_HEADERS_DIR`.
+By default, the OpenCL ICD Loader will look for OpenCL Headers in the `inc` directory.
+
+The OpenCL ICD Loader uses CMake for its build system.
+If CMake is not provided by your build system or OS pcakage manager, please consult the [CMake website](https://cmake.org).
+
+### Build and Install Directories
+
+A common convention is to place the `build` directory in the top directory of the repository and to place the `install` directory as a child of the `build` directory.
+The remainder of these instructions follow this convention, although you may place these directores in any location.
```
Comment (EwanC) [inline README.md]: `directores` -> `directories`

Comment (bashbaug) [reply README.md]: Fixed - thanks!

**Pull request merged**